src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,77 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+enum TodoStatus {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+type Props = {
+  setSortType: (status: TodoStatus) => void,
+  setQuery: (query: string) => void,
+  query: string,
+};
+
+export const TodoFilter: React.FC<Props> = (
+  { setSortType, setQuery, query },
+) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={(event) => {
+              let status;
+
+              switch (event.target.value) {
+                case 'active':
+                  status = TodoStatus.ACTIVE;
+                  break;
+                case 'completed':
+                  status = TodoStatus.COMPLETED;
+                  break;
+                default:
+                  status = TodoStatus.ALL;
+              }
+
+              setSortType(status);
```

Comment:
You can simplify this, it's not good practice to use `as` but here it will be useful
```suggestion
            onChange={(event) => {
              setSortType(event.taget.value as TodoStatus);
```